go/internal/feast/server/http_server.go

```diff
@@ -192,22 +221,86 @@ func (s *httpServer) getOnlineFeatures(w http.ResponseWriter, r *http.Request) {
 		writeJSONError(w, fmt.Errorf("Error decoding JSON request data: %+v", err), http.StatusInternalServerError)
 		return
 	}
+
 	var featureService *model.FeatureService
+	var entitiesProto = make(map[string]*prototypes.RepeatedValue)
+	var requestContextProto = make(map[string]*prototypes.RepeatedValue)
+	var fVList = make([]string, 0)
+
 	if request.FeatureService != nil {
 		featureService, err = s.fs.GetFeatureService(*request.FeatureService)
 		if err != nil {
 			logSpanContext.Error().Err(err).Msg("Error getting feature service from registry")
 			writeJSONError(w, fmt.Errorf("Error getting feature service from registry: %+v", err), http.StatusInternalServerError)
 			return
 		}
+		for _, fv := range featureService.Projections {
+			fVList = append(fVList, fv.Name)
+		}
+	} else if request.Features != nil && len(request.Features) > 0 {
+		log.Info().Msgf("request.Features %v", request.Features)
+		for _, featureName := range request.Features {
+			fvName, _, err := onlineserving.ParseFeatureReference(featureName)
+			if err != nil {
+				logSpanContext.Error().Err(err)
+				writeJSONError(w, fmt.Errorf("Error parsing feature reference %s", featureName), http.StatusBadRequest)
+				return
+			}
+			fVList = append(fVList, fvName)
+		}
+	} else {
+		logSpanContext.Error().Msg("No feature views or feature services specified in your request")
+		writeJSONError(w, errors.New("No feature views or feature services specified in your request"), http.StatusBadRequest)
+		return
+	}
+	requestSources, err2 := s.fs.GetRequestSources(fVList)
+	entityKeyTypeMap, err1 := s.fs.GetEntityKeyTypeMaps()
+	unifiedMap := make(map[string]prototypes.ValueType_Enum, 0)
+	if entityKeyTypeMap != nil && len(entityKeyTypeMap) > 0 {
+		for key, value := range entityKeyTypeMap {
+			unifiedMap[key] = value
+		}
 	}
-	entitiesProto := make(map[string]*prototypes.RepeatedValue)
-	for key, value := range request.Entities {
-		entitiesProto[key] = value.ToProto()
+	if requestSources != nil && len(requestSources) > 0 {
+		for key, value := range requestSources {
+			unifiedMap[key] = value
+		}
 	}
-	requestContextProto := make(map[string]*prototypes.RepeatedValue)
-	for key, value := range request.RequestContext {
-		requestContextProto[key] = value.ToProto()
+	if len(unifiedMap) > 0 {
+		if request.Entities != nil && len(request.Entities) > 0 {
```

Comment:
you can just use len(request.entities) > 0

Reply:
I added that condition in to check for the map, but I guess if i'm checking the length of the entity it's probably not needed. I'll remove it.